### ISSUE TITLE
[SF][A11y][AnalyticalTable]: [active content can't be read by JAWs]

### ISSUE THREAD

**Issue by JosephineSu** [SAP/ui5-webcomponents-react#5555]
### Describe the bug

Hi Colleagues,
There is an announcement issue reported by custom: AnalyticalTable content can't be read out by JAWS(not the latest version).
See below screenshot, suppose the Name column is clickable, when "Allen Best" is selected, only "Name" read out, which is not expected. We notice role="grid" has been used, and this will disrupt screen reading software.
![image](https://github.com/SAP/ui5-webcomponents-react/assets/38159212/540442d3-5087-4581-886b-1b488c889e02)
![image](https://github.com/SAP/ui5-webcomponents-react/assets/38159212/6d37893d-846a-40eb-a0eb-78ae8e51bcec)

### Expected Behaviour

When active item got focus, it should be announced instead of announce column name.

### UI5 Web Components for React Version

1.25.1

### UI5 Web Components Version

1.22.0

### Browser

Chrome

### Additional Context
transfer from discussion
https://github.com/SAP/ui5-webcomponents-react/discussions/5511

### Organization

SF

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Hi @JosephineSu 

the linked PR will fix the issue. It turns out that we had a bug concerning non-interactive tables where the `aria-label` wasn't set correctly. As I mentioned in the discussion, the `grid` role shouldn't be a problem as grid tables are widely supported and even SAPUI5 uses these roles for some of their tables.
